lessons/Early-Math/variables/langs/en-us/pages/variables-code2.adoc

```diff
@@ -0,0 +1,71 @@
+
+[.FillVerticalSpace,cols="1, 1", stripes="none", options="header"]
+|===
+
+| Notice | Wonder
+|
+|
+
+|===
+
+@n Based on what you see in the Definitions Window (left side), predict what will happen when you hit "Run" by circling your choice below.
+
+@vspace{1ex}
+
+[cols="^.^1,^.^1,^.^1,^.^1", stripes="none"]
+|===
+
+| Looks shipshape, all 4 tests passed, mate!
+| 3 tests passed, 1 test failed
+| 2 tests passed, 2 tests failed
+| 1 test passed, 3 tests failed
+
+|===
+
+@n Explain how you made your prediction (above). @fitb{}{}
+
+@fitb{}{}
+
+@fitb{}{}
+
+@n Click "Run". Were you correct? @fitb{5em}{} Explain. @fitb{}{@ifsoln{After hitting "Run", we discover that 3 tests passed and 1 test (example 3) failed.}}
+
+@fitb{}{@ifsoln{Test 3 fails because @math{(1 - 1) - 1}, which evaluates to -1, is not equivalent to @math{1 - (1 - 1)}, which evaluates to 0.}}
+
+@fitb{}{}
+
+@n Give four sets of values for `a`, `b` and `c` that will cause _both_ example 3 and example 4 to fail. Try them out!
+
+@hspace{1em} a = @fitb{2em}{} @hspace{1em} b = @fitb{2em}{} @hspace{1em} c = @fitb{2em}{} @hspace{6em}
+a = @fitb{2em}{} @hspace{1em} b = @fitb{2em}{} @hspace{1em} c = @fitb{2em}{} @hspace{2em}
+
+@hspace{1em} a = @fitb{2em}{} @hspace{1em} b = @fitb{2em}{} @hspace{1em} c = @fitb{2em}{} @hspace{6em}
+a = @fitb{2em}{} @hspace{1em} b = @fitb{2em}{} @hspace{1em} c = @fitb{2em}{} @hspace{2em}
+
+@n Are there any sets of values for `a`, `b`, and `c` that will cause example 1 or example 2 to fail? If so, list them here: @fitb{}{@ifsoln{No.}}
+
+@n Are any of the examples true *every time, no matter what values we use for a and b*? If so, which ones? @fitb{}{@ifsoln{No.}}
+
+@n Are any of the examples true *some of the time, depending on what values we use for a and b*? If so, which ones? @fitb{}{@ifsoln{Yes, examples 3 and 4.}}
+
+@n Deena and Marcus want to edit the code of their starter file in their Definitions Window so that when they hit "Run", all four tests pass. They each have a different idea.
+
+- Marcus suggests leaving a, b, and c how they are (equal to 1) and instead changing the `true` in line 17 to say `false`.
+- Deena thinks they should change the values of the variables so that a, b, and c are each equal to zero.
+
+Who do you agree with? _Before deciding, feel free to test out Marcus's idea and Deena's idea._ @fitb{10em}{} Explain. @fitb{}{}
+
+@fitb{}{@ifsoln{Marcus's idea will work, but Deena's will not. When we test Deena's idea, examples 1, 2, and 3 pass, but example 4 fails }}
+
+@fitb{}{@ifsoln{because we cannot divide by zero.}}
+
+@fitb{}{}
+
+@star Janae dislikes Marcus's solution. She points to his code on line 17 and says, "This example is "false", so it has to be *wrong*. We can't leave it." How would you respond? @fitb{}{@ifsoln{Boolean-producing functions _should sometimes return false_! Just because an expression produces `false`}}
```

Comment:
This misconception is always a PITA for teachers to sort through. Another reason why I'm thinking we should get rid of `true` and `false` in these examples

Reply:
makes sense.